pkg/maintenance/maintenance_test.go

```diff
@@ -0,0 +1,169 @@
+package maintenance
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/managed-upgrade-operator/util/mocks"
+	amSilence "github.com/prometheus/alertmanager/api/v2/client/silence"
+	amv2Models "github.com/prometheus/alertmanager/api/v2/models"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Alert Manager Maintenance Client", func() {
+	var (
+		mockCtrl              *gomock.Controller
+		mockKubeClient        *mocks.MockClient
+		silenceClient         *MockAlertManagerSilencer
+		testComment           = "test comment"
+		testOperatorName      = "managed-upgrade-operator"
+		testCreatedByOperator = testOperatorName
+		testCreatedByTest     = "Tester the Creator"
+		testNow               = strfmt.DateTime(time.Now().UTC())
+		testEnd               = strfmt.DateTime(time.Now().UTC().Add(90 * time.Minute))
+		testVersion           = "V-1.million.25"
+		testGettableSilences  amv2Models.GettableSilences
+
+		// Create test silence NOT created by the operator
```

Comment:
Minor nit, this comment seems to indicate the opposite of what's being created.

Reply:
So my comment is incorrect. This was a validation that the operator will only retrieve maintenance's created by the operator. The actual test got moved to another spot in the file. Ill ammend